### PR TITLE
Configure pipeline jobs to fail if there are security vulnerabilities

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2317

### Change description

Configure pipeline jobs to fail if there are security vulnerabilities

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,6 +17,7 @@ build:
 	mvn versions:set -DnewVersion=$(version) -DgenerateBackupPoms=false
 	mvn package -DskipTests=true
 	cp ./target/$(artifact_name)-$(version).jar ./$(artifact_name).jar
+	mvn org.owasp:dependency-check-maven:check -DassemblyAnalyzerEnabled=false
```
Comment (mwestacott) [inline Makefile:20]: I don't think this line should be added. It's line number 62 that needs changing, to add the flag that causes a build failure, as per comments in lines above. I'd probably go for the `-DfailBuildOnAnyVulnerability=true` setting.

File: Makefile [hunk 2]
```diff
@@ -18,6 +18,7 @@ build:
 	mvn package -DskipTests=true
 	cp ./target/$(artifact_name)-$(version).jar ./$(artifact_name).jar
 
+
```
Comment (mwestacott) [inline Makefile:21]: I would remove this new, empty blank line.

File: Makefile (hunk 1 above)
Comment (HatiDev) [reply Makefile]: I have removed the line and added the check after line 62

File: Makefile [hunk 3]
```diff
@@ -59,4 +60,5 @@ security-check:
     # we fix the vulnerabilities by changing the flags on the dependency check
     # e.g.
     # -DfailBuildOnCVSS=0 or -DfailBuildOnAnyVulnerability=true
-	mvn org.owasp:dependency-check-maven:check -DassemblyAnalyzerEnabled=false
\ No newline at end of file
+	mvn org.owasp:dependency-check-maven:check -DassemblyAnalyzerEnabled=false
```
Comment (mwestacott) [inline Makefile:63]: So (I think) it's just this line that needs modifying to add the `-DfailBuildOnAnyVulnerability=true` flag at the end. No new invocation of the dependency checker is needed, so the new line you added can go. Also the three comment lines above, as they don't apply any more.

File: Makefile (hunk 1 above)
Comment (mwestacott) [reply Makefile]: 👍 thanks. Please see new comments added below.

File: Makefile (hunk 2 above)
Comment (HatiDev) [reply Makefile]: removed

File: Makefile (hunk 3 above)
Comment (HatiDev) [reply Makefile]: comments removed and line edited

File: Makefile [hunk 4]
```diff
@@ -55,8 +55,5 @@ sonar-pr-analysis:
 
 .PHONY: security-check
 security-check:
-    # This target currently doesn't fail a build if there are reported vulnerabilities. This will need to be changed when
-    # we fix the vulnerabilities by changing the flags on the dependency check
-    # e.g.
-    # -DfailBuildOnCVSS=0 or -DfailBuildOnAnyVulnerability=true
-	mvn org.owasp:dependency-check-maven:check -DassemblyAnalyzerEnabled=false
\ No newline at end of file
+
```
Comment (mwestacott) [inline Makefile:58]: Please remove this blank line.

**Review by mwestacott — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**